rpaas/storage.py

```diff
@@ -43,18 +43,26 @@ def remove_hc(self, name):
 
     def store_task(self, name):
         try:
-            self.db[self.tasks_collection].insert({'_id': name})
+            if name.__class__ is dict:
+                self.db[self.tasks_collection].insert(name)
+            else:
+                self.db[self.tasks_collection].insert({'_id': name})
         except pymongo.errors.DuplicateKeyError:
             raise DuplicateError(name)
 
-    def remove_task(self, name):
-        self.db[self.tasks_collection].remove({'_id': name})
+    def remove_task(self, query):
+        self.db[self.tasks_collection].remove(query)
 
-    def update_task(self, name, task_id):
-        self.db[self.tasks_collection].update({'_id': name}, {'$set': {'task_id': task_id}})
+    def update_task(self, name, task_id_or_spec):
+        if task_id_or_spec.__class__ is dict:
+            self.db[self.tasks_collection].update({'_id': name}, {'$set': task_id_or_spec})
+        else:
+            self.db[self.tasks_collection].update({'_id': name}, {'$set': {'task_id': task_id_or_spec}})
 
-    def find_task(self, name):
-        return self.db[self.tasks_collection].find_one({'_id': name})
+    def find_task(self, query):
+        if query.__class__ is dict:
+            return self.db[self.tasks_collection].find(query)
+        return self.db[self.tasks_collection].find_one(query)
```

Comment:
A better way here is to change the `find_task` to return a list and change the code that uses `find_task` to get the first value of find_task